src/app/exercise/exercise.page/state/fsrs-exercise.ts

```diff
@@ -58,8 +58,10 @@ export class QuestionCardsCollection<GAnswer extends string> {
   }
 
   remove(savedQuestion: QuestionCard<GAnswer>) {
-    this._savedQuestions = this._savedQuestions.filter(
-      (q) => q.question.id !== savedQuestion.question.id,
+    this._savedQuestions = this._savedQuestions.filter((q) =>
+      q.question.id
+        ? q.question.id !== savedQuestion.question.id
+        : !isEqual(q.question, savedQuestion.question),
```

Comment:
Crucial fix that caused learn mode to not work at all if ids didn't exist on questions.